src/main/scala/org/dia/algorithms/mcc/MCSUtils.scala

```diff
@@ -87,30 +87,24 @@ object MCSUtils {
         }
     }
 
-    /** Get the data from the nodes
-     * @param edgeArray Array[(String, String)] representing the edgelist
+
+    /**
+     * Get the data from the nodes in the edge
+     * @param edge Current MCCEdge
      * @param MCSNodeMap mutable.HashMap[String, MCCNode] representing the map of each node metadata
      * @param lats Array[Double] representing the latitudes
      * @param lons Array[Double] representing the longitudes
-     * @param tightestBox Boolean to use tightest box around data.
+     * @param tightestBox Boolean to use tightest box around data
      */
-    def get_all_node_data(edgeArray: Array[(String, String)], MCSNodeMap: mutable.HashMap[String, MCCNode],
-      lats: Array[Double], lons: Array[Double], tightestBox: Boolean): Unit = {
-        edgeArray.map({ e =>
-            var seenNodes = new mutable.ListBuffer[String]()
-            val first = e._1
-            if (!seenNodes.contains(first)) {
-                seenNodes += first
-                val fNode = MCSNodeMap(first)
-                extract_masked_data(fNode, lats, lons, tightestBox)
-            }
-            val second = e._2
-            if (!seenNodes.contains(second)) {
-                seenNodes += second
-                val sNode = MCSNodeMap(second)
-                extract_masked_data(sNode, lats, lons, tightestBox)
-            }
-        })
+    def get_node_data(edge: MCCEdge, MCSNodeMap: mutable.HashMap[String, MCCNode],
```

Comment:
This function does alot of things that is not so apparent when calling the function.
Namely, it also writes netcdf files, because of the call to extract_masked_data(), which also calls netcdf_write. The responsibilities of these functions should be decoupled from each other.